src/components/popover/tab/keypress/keypress-tab.component.ts

```diff
@@ -16,6 +16,8 @@ export class KeypressTabComponent implements OnChanges, Tab {
     @Input() defaultKeyAction: KeyAction;
     @Input() longPressEnabled: boolean;
 
+    @Output() validAction = new EventEmitter();
```

Comment:
Why have you created a new event? What don't you use `keyActionValid` function of the `Tab` interface? It is even easily accesible from popover.